### ISSUE TITLE
crm refcodes not visible when entered to an existing partner

### ISSUE THREAD

**Issue by dulhaver** [dreispt/odoo-addons#1]
behaviour:
- the "Contact Reference" only shows on kanban and list views when it is being entered at the moment a partner is created. 
- If you add a Contact Reference to an existing partner it does not show, nor is searchable

expected behaviour:
- as soon you enter a contact reference to a Supplier or Customer. 
- It shows in the views of Suppliers or Customer. No matter whether the partner is existing or created fresh.

affected version:
- 8.0

reproduce by:
- creating a Customer or Supplier without adding a "Contact Reference". 
- Save it
- press on Edit again and add the Contact Reference
- check the kanban/list view

restarting the odoo server, or the browser view [F5] does not solve this issue

EDIT: also, when you change the Customer reference to an existing partner the new code only shows on the Form view. On the kanban view as well as the Tree View the originally defined code remains. The new code is not found by the search/filter either

**Comment by dulhaver** [comment]
am I the only one with this problem ?

**Comment by dreispt** [comment]
I'm sorry, this went under my radar. I'll check on this.

**Comment by dulhaver** [comment]
great

**Comment by dreispt** [comment]
@vrms Sorry for the long delay.

I checked this and it is working as expected (by me): 
- The code is displayed along with the name in other documents/records, but not in the Partners list and Kanban
- However, it is searchable - searching by name also searches by code

If you need to see the Code field visible in the Tree/Kanban views, you should add it.
Maybe the module could be changed to also do that...

**Comment by dreispt** [comment]
@vrms I'm sorry, forget the previous message: I was working on this and I didn't reproduce the issue.

Steps taken:
1. Create a new Partner "ACME" with Contact Reference "XYZ" and Save.
2. On the list view, search for "XYZ": ACME is found as is displayed as "[XYZ] ACME"
3. Switch to Kanban view: ACME's name is also displayed as "[XYZ] ACME"

Am I missing something?

**Comment by dulhaver** [comment]
yes, maybe I was not clear enough. I am talking about partner that have been saved either without a Contact Reference or a case where you want to change the contact reference (maybe because you typed it wrong) for an existing Partner
1. create a new partner and do not set the Contact reference
2. save
3. damn ... you forgot to put the Contact reference !!!
4. reopen the just created partner, add Contact Reference & save
   ... the contact reference does not appear in the List- nor KanBan view

.... if you change Contact Reference in an Existing Partner the KanBan & List view continue to show the 'old' Reference. Only the Form view shows the adjusted one

**Comment by dulhaver** [comment]
@dreispt: could you reinact the error?

**Comment by dreispt** [comment]
No, I couldn't.
But I'll try a second time.

**Comment by dulhaver** [comment]
I still face this issue ... have you had a chance to verify it's existence yet?
